### PR TITLE
Split createAndAddAccount and use a lock while modifying the accounts

### PR DESCRIPTION
**Pull request by benma**

### CODE REVIEW

**Comment by benma** [comment]
> backend/backend.go:340:3: (*Backend).createAndAddAccount - result 0 (error) is always nil (unparam)
) error {

This is the kind of linting I like!

**Review by x1ddos — COMMENTED** [review]
Are there any tests for all these account init stuff? Not sure I can see any. More changes are coming to this - now would be a very good time for tests.

File: backend/backend.go [hunk 1]
```diff
@@ -299,48 +299,45 @@ func (backend *Backend) emitAccountsStatusChanged() {
...
+		})
+		return nil
+	})
+}
 
+// The accountsLock must be held when calling this function.
+func (backend *Backend) createAndAddAccount(
+	coin coin.Coin,
+	code string,
+	name string,
```
[the remark this answers is not in the document]
Comment (benma) [reply backend/backend.go:338]: Those will probably be reformed/restrucuted/renamed in the next PRs, will make sure to document them then.

File: backend/config/config.go [hunk 2]
```diff
@@ -336,10 +336,13 @@ func (config *Config) AccountsConfig() AccountsConfig {
 	return config.accountsConfig
 }
 
-// SetAccountsConfig sets and persists the accounts config.
-func (config *Config) SetAccountsConfig(accountsConfig AccountsConfig) error {
+// ModifyAccountsConfig modifies the accounts config. Use this for thread-safety.
```
[the remark this answers is not in the document]
Comment (benma) [reply backend/config/config.go]: Done

File: backend/backend.go [hunk 3]
```diff
@@ -309,32 +309,34 @@ func (backend *Backend) createAndAddAccount(
...
+			accountsConfig.Accounts = append(accountsConfig.Accounts, config.Account{
+				CoinCode:       coin.Code(),
+				Code:           code,
+				Name:           name,
+				Configurations: configurations,
+			})
+			return nil
 		})
-		if err := backend.config.SetAccountsConfig(accountsConfig); err != nil {
+		if err != nil {
```
[the remark this answers is not in the document]
Comment (benma) [reply backend/backend.go]: I think this might be outdated?

**Review by x1ddos — APPROVED** [review]
I guess LGTM but it feels like some tests are really needed here. It's almost impossible to test various cases manually.

**Pull request merged**